core_types/src/entry/agent.rs

```diff
@@ -1,43 +1,104 @@
-use cas::content::{AddressableContent, Content};
+use cas::content::{Address, AddressableContent, Content};
 use entry::{Entry, ToEntry};
 use entry_type::EntryType;
 use json::JsonString;
 
-/// Object holding an Agent's identity.
-#[derive(Clone, Debug, PartialEq)]
-pub struct Identity(Content);
+use std::convert::TryFrom;
+
+use super::super::error::HolochainError;
+use reed_solomon::{Decoder, Encoder};
+
+const PARITY_LEN: usize = 2;
+
+pub type Identity = String;
+
+#[derive(Clone)]
+pub struct IdentityBuffer([u8; 64]);
+
+impl IdentityBuffer {
+    pub fn parse(s: &str) -> Result<IdentityBuffer, HolochainError> {
+        let s = s.replace("-", "+").replace("_", "/");
+        let s = base64::decode(&s)?;
+        let dec = Decoder::new(PARITY_LEN);
+        let dec = *dec.correct(s.as_slice(), None)?;
+        Ok(IdentityBuffer(array_ref![dec, 0, 64].clone()))
+    }
+
+    pub fn get_sig(&self) -> &[u8; 32] {
+        array_ref![self.0, 0, 32]
+    }
 
-impl From<Identity> for JsonString {
-    fn from(identity: Identity) -> JsonString {
-        identity.0
+    pub fn get_enc(&self) -> &[u8; 32] {
+        array_ref![self.0, 32, 32]
     }
 }
 
-impl From<JsonString> for Identity {
-    fn from(json_string: JsonString) -> Identity {
-        Identity(json_string)
+#[derive(Clone, Debug, PartialEq)]
+pub struct Agent(Identity);
+
+impl Agent {
+    pub fn generate_fake(s: &str) -> Self {
+        let mut s = s.to_string();
+        while s.len() < 85 {
+            s.push_str("+");
+        }
+        s.push_str("A==");
+        let s = base64::decode(&s)
+            .expect("could not decode the generated fake base64 string - use the base64 alphabet");
+        let s = IdentityBuffer(array_ref![s, 0, 64].clone());
+        Agent::render(&s)
```

Comment:
What exactly is happening here? Seems like you are creating something that is a syntactilly correct (public?) key from a string? I understood those names in the tests ("Terence", "Florence") to be the nick associated with a key.